src/components/toast/toast.tsx

```diff
@@ -0,0 +1,205 @@
+import {
+  Component,
+  ComponentInterface,
+  h,
+  Prop,
+  Method,
+  Element,
+  Event,
+  EventEmitter,
+} from "@stencil/core";
+
+export type ActionType = "button" | "icon";
+
+export type VariantType = "system" | "error" | "success" | "warning";
+
+export type ButtonActionType = "close" | "custom";
+
+export type CreateToastType = {
+  toastContainer: HTMLToastContainerElement;
+  toastElement: HTMLBdsToastElement;
+  actionType: ActionType;
+  buttonAction: ButtonActionType;
+  buttonText?: string;
+  icon?: string;
```

Comment:
Is `icon` a string?

Reply:
yea, it will receive the name of the icon which will be used in the <bds-icon> component